app/Importer/LicenseImporter.php

```diff
@@ -73,12 +73,21 @@ public function createLicenseIfNotExists(array $row)
                 $asset = Asset::where('asset_tag', $asset_tag)->first();
                 $targetLicense = $license->freeSeat();
                 if ($checkout_target) {
-                    $targetLicense->assigned_to = $checkout_target->id;
-                    $targetLicense->user_id = Auth::id();
-                    if ($asset) {
-                        $targetLicense->asset_id = $asset->id;
+                    try{
+                        $targetLicense->assigned_to = $checkout_target->id;
+                        $targetLicense->user_id = Auth::id();
+                        if ($asset) {
+                            $targetLicense->asset_id = $asset->id;
+                        }
+                        $targetLicense->save();
+                    } catch (\Exception $ex){
+                        if($ex->getMessage() === "Creating default object from empty value"){
```

Comment:
This feels like a very, very clever hack to try and catch that there aren't enough seats. And you know how I feel about things that are 'clever' :)

Can't we look at the output of the `freeSeat()` method above, and if it's `null`, we can give this type of error? That seems more straightforward at least.

Reply:
As a side-stylistic note, I generally don't like "using exceptions for flow control." A license running out of seats is more of an everyday occurrence, so we probably shouldn't need a try/catch scenario.

Reply:
Change was made. I think the new commit already takes care of both comments. Thanks a lot for your input!!